<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>804</width>
    <height>597</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <widget class="QWidget" name="imageWidget" native="true">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>441</width>
      <height>541</height>
     </rect>
    </property>
   </widget>
   <widget class="QWidget" name="containerWidget" native="true">
    <property name="geometry">
     <rect>
      <x>470</x>
      <y>10</y>
      <width>211</width>
      <height>541</height>
     </rect>
    </property>
    <widget class="QWidget" name="contrastWidget" native="true">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>80</y>
       <width>201</width>
       <height>101</height>
      </rect>
     </property>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>181</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>Contrast</string>
      </property>
     </widget>
     <widget class="QSlider" name="slider_contrast">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>171</width>
        <height>21</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QLabel" name="label_4">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>60</y>
        <width>181</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>Brightness</string>
      </property>
     </widget>
     <widget class="QSlider" name="slider_brightness">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>80</y>
        <width>171</width>
        <height>21</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="blurWidget" native="true">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>201</width>
       <height>71</height>
      </rect>
     </property>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>181</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>Kernel size for blurring</string>
      </property>
     </widget>
     <widget class="QSlider" name="slider_blur">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>50</y>
        <width>171</width>
        <height>21</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QCheckBox" name="gaussianBox">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>0</y>
        <width>92</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Gaussian</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="rotationWidget" native="true">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>350</y>
       <width>201</width>
       <height>91</height>
      </rect>
     </property>
     <widget class="QPushButton" name="buttonRotateM90">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>10</y>
        <width>89</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>Rotate - 90°</string>
      </property>
     </widget>
     <widget class="QPushButton" name="buttonRotateP90">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>10</y>
        <width>89</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>Rotate + 90°</string>
      </property>
     </widget>
     <widget class="QPushButton" name="customRotate">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>60</y>
        <width>61</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>Rotate</string>
      </property>
     </widget>
     <widget class="QLabel" name="labelRotate">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>40</y>
        <width>181</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>Custom rotation</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="customRotateInput">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>60</y>
        <width>113</width>
        <height>25</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="widget" native="true">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>440</y>
       <width>201</width>
       <height>101</height>
      </rect>
     </property>
     <widget class="QLabel" name="labelResize">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>0</y>
        <width>181</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>Resize</string>
      </property>
     </widget>
     <widget class="QLabel" name="labelCrop">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>50</y>
        <width>181</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>Crop inputs</string>
      </property>
     </widget>
     <widget class="QPushButton" name="cropButton">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>70</y>
        <width>61</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>Crop</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="cropInputX">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>70</y>
        <width>51</width>
        <height>25</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="cropInputY">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>70</y>
        <width>51</width>
        <height>25</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="resizeW">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>51</width>
        <height>25</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="resizeButton">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>20</y>
        <width>61</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>Resize</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="resizeH">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>20</y>
        <width>51</width>
        <height>25</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="contrastWidget_2" native="true">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>190</y>
       <width>201</width>
       <height>71</height>
      </rect>
     </property>
     <widget class="QLabel" name="erosionLabel">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>181</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>Erosion</string>
      </property>
     </widget>
     <widget class="QSlider" name="erosionBar">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>50</y>
        <width>171</width>
        <height>21</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QRadioButton" name="erosionRadio1">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>30</y>
        <width>61</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Cross</string>
      </property>
     </widget>
     <widget class="QRadioButton" name="erosionRadio0">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>61</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Rect</string>
      </property>
     </widget>
     <widget class="QRadioButton" name="erosionRadio2">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>30</y>
        <width>71</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Ellipse</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="contrastWidget_3" native="true">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>270</y>
       <width>201</width>
       <height>71</height>
      </rect>
     </property>
     <widget class="QLabel" name="dilationLabel">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>181</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>Dilation</string>
      </property>
     </widget>
     <widget class="QSlider" name="dilationBar">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>50</y>
        <width>171</width>
        <height>21</height>
       </rect>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
     <widget class="QRadioButton" name="dilationRadio1">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>30</y>
        <width>61</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Cross</string>
      </property>
     </widget>
     <widget class="QRadioButton" name="dilationRadio0">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>61</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Rect</string>
      </property>
     </widget>
     <widget class="QRadioButton" name="dilationRadio2">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>30</y>
        <width>71</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Ellipse</string>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>804</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFiles">
    <property name="title">
     <string>Files</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
   </widget>
   <widget class="QMenu" name="menuSettings">
    <property name="title">
     <string>Settings</string>
    </property>
    <addaction name="actionLanguage"/>
   </widget>
   <widget class="QMenu" name="menuAbout">
    <property name="title">
     <string>About</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <widget class="QMenu" name="menuImages">
    <property name="title">
     <string>Images</string>
    </property>
    <addaction name="actionReset"/>
    <addaction name="separator"/>
    <addaction name="actionToGrey"/>
    <addaction name="actionDetectEdges"/>
    <addaction name="separator"/>
    <addaction name="actionPanorama"/>
   </widget>
   <addaction name="menuFiles"/>
   <addaction name="menuSettings"/>
   <addaction name="menuAbout"/>
   <addaction name="menuImages"/>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>RightToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionOpen"/>
   <addaction name="actionSave"/>
   <addaction name="separator"/>
   <addaction name="actionReset"/>
   <addaction name="separator"/>
   <addaction name="actionToGrey"/>
   <addaction name="actionDetectEdges"/>
   <addaction name="separator"/>
   <addaction name="actionPanorama"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionOpen">
   <property name="text">
    <string>Open Image</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
  </action>
  <action name="actionLanguage">
   <property name="text">
    <string>Language</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionToGrey">
   <property name="text">
    <string>To grey</string>
   </property>
  </action>
  <action name="actionReset">
   <property name="text">
    <string>Reset</string>
   </property>
  </action>
  <action name="actionDetectEdges">
   <property name="text">
    <string>Edge Detection</string>
   </property>
  </action>
  <action name="actionPanorama">
   <property name="text">
    <string>Panorama</string>
   </property>
   <property name="toolTip">
    <string>Open the assets folder</string>
   </property>
  </action>
  <action name="actionBlur">
   <property name="text">
    <string>Blur</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>actionOpen</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>open()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionToGrey</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>imageToGrey()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>401</x>
     <y>298</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionReset</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>reset()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>401</x>
     <y>298</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionDetectEdges</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>detectEdges()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>401</x>
     <y>298</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionPanorama</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>createPanorama()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>401</x>
     <y>298</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>open()</slot>
  <slot>reset()</slot>
  <slot>imageToGrey()</slot>
  <slot>undo()</slot>
  <slot>redo()</slot>
  <slot>detectEdges()</slot>
  <slot>createPanorama()</slot>
  <slot>createBlurImage()</slot>
 </slots>
</ui>
